lib/enigma.rb

```diff
@@ -1,8 +1,40 @@
+require 'date'
 class Enigma
+  def rand_key
+    rand(99999)
+  end
 
-  def pad(number)
-    pad_num = number.to_s
+  def zero_pad(key)
+    pad_num = key.to_s
     pad_num.insert(0, "0") until pad_num.size == 5
     pad_num
   end
+
+  def shift_keys(key)
+    keys = []
+    key.chars.each_with_index do |char, index|
+      next if index == 4
+      keys << char + key[index + 1]
+    end
+    keys
+  end
```

Comment:
"shift_keys" will be taking our key, after zero_pad, and turning it into an array of two digit numbers by adding combining each number with the next number in the array. 
Needed to use each_with_index to be able to access the next number in the array. Also had to add the "next if" line to tell the enumerable to stop before the next number becomes "nil".